tools/precommit.py

```diff
@@ -117,28 +119,46 @@ def build(buildtype, args=[]):
     ex.check_run_cmd('./tools/build.py', ['--buildtype=' + buildtype] + args)
 
 
+def get_os_dependency_exclude_module(exclude_module):
+    os_dependency_module = {}
+    all_module = set(exclude_module['all'])
+    for os_name in exclude_module.keys():
+        if not os_name == 'all':
+            os_dependency_module[os_name] = \
+              list(all_module | set(exclude_module[os_name]))
+    return os_dependency_module
+
+
 option = parse_option()
 config = get_config()
-if len(config['module']['exclude']) > 0 :
-    include_module = ','.join(config['module']['exclude'])
-    include_module = ['--iotjs-include-module=' + include_module]
-else:
-    include_module = []
+os_dependency_module = \
+  get_os_dependency_exclude_module(config['module']['exclude'])
+
+# Excluded modules are also included in the build test.
+# Travis will test all implemented modules.
```

Comment:
Honestly, the deeper I dig into the semantics of the "exclude" list in build.config, build.py, and precommit.py, the more I get confused. (E.g.,`--iotjs-exclude-module` is never used anywhere. The "exclude" option in build.config is only used by build.py to check whether they are not required by included modules. However, by precommit.py, they are turned into included modules for build tests.) It really sounds like a misnomer to me, or two concepts mixed into one (not sure yet which one is the case).

I acknowledge that the core of this logic is inherited, so I'm not insisting to fix it in this patch (if you don't want to), as it might require a bit bigger or differently scoped refactoring. But please do let me know whether you plan to tackle it (now or later). If you'd like to leave it for someone else, I'll open a discussion issue for this topic.